packages/@aws-cdk/aws-neptune-alpha/lib/cluster.ts

```diff
@@ -644,7 +644,9 @@ export class DatabaseCluster extends DatabaseClusterBase implements IDatabaseClu
       // things in the right order.
       instance.node.addDependency(internetConnectivityEstablished);
 
-      instance.applyRemovalPolicy(props.removalPolicy, {
+      // Neptune instance's default deletion policy should be DESTROY to align with the CFN behaviour and
+      // to allow customers to set cluster's RemovalPolicy to SNAPSHOT since instance doesn't support SNAPSHOT.
+      instance.applyRemovalPolicy(RemovalPolicy.DESTROY, {
```

Comment:
Won't this set removal policy to DESTROY irrespective of what is set by the user?

Reply:
Yes, that is the intention.

This is the construct for cluster and users trying to set SNAPSHOT deletion policy for cluster are not able to because it gets inherited by the instance(s) created via cluster prop at this line and results in the error `Resource type AWS::Neptune::DBInstance does not support Snapshot DeletionPolicy` thus preventing users from using that deletion policy.

The instance(s) auto-created via the cluster prop `instances?` should use the DESTROY removal policy as it is the default behaviour in CFN.
If users want to customize the instance removal policy (instead of using default), they are expected to use the [DatabaseInstance](https://docs.aws.amazon.com/cdk/api/v2/docs/@aws-cdk_aws-neptune-alpha.DatabaseInstance.html) construct.